src/connection.rs

```diff
@@ -442,6 +448,55 @@ impl Connection {
         Ok(self.run_command(command).await?.optional_string())
     }
 
+    ///Pop a value from one of the lists from the left side.
+    ///Block timeout seconds when there are no values to pop (timeout=0 means infinite)
+    ///# Return value
+    ///* `Ok(Some((list,value)))`: name of the list and corresponding value
+    ///* `Ok(None)`: timeout (no values)
+    ///* `Err(err)`: there was an error
+    pub async fn blpop<K>(&mut self, lists: &[K], timeout: u32) -> Result<Option<Vec<Vec<u8>>>>
+    where
+        K: AsRef<[u8]>,
+    {
+        self.blpop_brpop(lists, timeout, "BLPOP").await
+    }
+
+    ///Pop a value from one of the lists from the right side.
+    ///Block timeout seconds when there are no values to pop (timeout=0 means infinite)
+    ///# Return value
+    ///* `Ok(Some((list,value)))`: name of the list and corresponding value
+    ///* `Ok(None)`: timeout (no values)
+    ///* `Err(err)`: there was an error
+    pub async fn brpop<K>(&mut self, lists: &[K], timeout: u32) -> Result<Option<Vec<Vec<u8>>>>
+    where
+        K: AsRef<[u8]>,
+    {
+        self.blpop_brpop(lists, timeout, "BRPOP").await
+    }
+
+    ///blpop and brpop common code
+    async fn blpop_brpop<K>(
```

Comment:
This function would have to be changed if we are to return a nil value when getting a nil array.

Reply:
It's done:
* 4ea8804: suggested **Connection::parse_array()** adaption to return **Value::Nil**
* 4591364: adapted **Connection::blpop_brpop()** to support **Value::Nil**